### PR TITLE
Fix Basic Auth Configuration for Multiple Paths

### PR DESCRIPTION
**Pull request by rathahin**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

The existing implementation of basic authentication works partially when multiple routes are specified. The `.some()` method used in the code causes it to return as soon as one matching path is found, leading to incorrect behavior for additional paths. 

This change replaces the `.some()` method with individual variables to track the `exclude` and `include` lists. It uses `.some()` to update these variables and then checks them to decide whether to block the request.

This fix ensures that all specified paths in both the `exclude` and `include` arrays are correctly checked. It provides a clearer and more accurate way to handle basic authentication for multiple routes.

Resolves: #[Issue Number]

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes (if not applicable, please state why)

### CODE REVIEW

File: pnpm-lock.yaml
Comment (Baroshem) [inline pnpm-lock.yaml:1]: issue: this is not needed for sure ;)

**Review by Baroshem — CHANGES_REQUESTED** [review]
One small issue. If you revert it, it should be ready to review and further merge :)

File: pnpm-lock.yaml
Comment (rathahin) [reply pnpm-lock.yaml:1]: Thank you for pointing out the issue with `pnpm-lock.yaml`. I understand it's not needed and I'll revert it back to its original state. Once that's done, it should be ready for your review and potential merge.

**Review by Baroshem — APPROVED** [review]
Thanks!

**Comment by Baroshem** [comment]
Hey @rathahin 

Could you please adjust the unit tests to fix the CI?

**Comment by Baroshem** [comment]
Thanks @rathahin 💚

I will merge it now to the rc branch, conduct some tests and will release a new RC version today :)

**Comment by Baroshem** [comment]
Thank you @rathahin ! 💚

**Pull request merged**